base/iterator.jl

```diff
@@ -83,6 +103,37 @@ immutable Zip{I, Z<:AbstractZipIterator} <: AbstractZipIterator
     a::I
     z::Z
 end
+
+"""
+    zip(iters...)
+
+For a set of iterable objects, returns an iterable of tuples, where the `i`th tuple contains
+the `i`th component of each input iterable.
+
+Note that [`zip`](:func:`zip`) is its own inverse: `collect(zip(zip(a...)...)) == collect(a)`.
+
+```jldoctest
+julia> a = 1:5
+1:5
+
+julia> b = ["e","d","b","c","a"]
+5-element Array{String,1}:
+ "e"
+ "d"
+ "b"
+ "c"
+ "a"
+
+julia> c = zip(a,b)
+Base.Zip2{UnitRange{Int64},Array{String,1}}(1:5,String["e","d","b","c","a"])
+
+julia> length(c)
+5
+
+julia> first(c)
+(1,"e")
+```
+"""
 zip(a, b, c...) = Zip(a, zip(b, c...))
```

Comment:
(Yes, when it can't find a doc matching the given arguments it'll just display them all.)